Observer/SalesQuoteProductAddAfter.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+
+namespace Klaviyo\Reclaim\Observer;
+
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Event\ObserverInterface;
+use Magento\Catalog\Model\CategoryFactory;
+use Magento\Checkout\Model\Session;
+
+class SalesQuoteProductAddAfter implements ObserverInterface
+{
+    /**
+     * Magento Category Factory
+     * @var CategoryFactory $_categoryFactory
+     */
+    protected $_categoryFactory;
+
+    /**
+     * Magento Checkout Session data
+     * @var Session $_checkoutSession
+     */
+    protected $_checkoutSession;
+
+    /**
+     * SalesQuoteProductAddAfter constructor.
+     * @param CategoryFactory $categoryFactory
+     * @param Session $checkoutsession
+     */
+    public function __construct(
+        CategoryFactory $categoryFactory,
+        Session $checkoutsession
+    )
+    {
+        $this->_categoryFactory = $categoryFactory;
+        $this->_checkoutSession = $checkoutsession;
+    }
+
+    public function execute(Observer $observer)
+    {
+        $quote = $observer->getData('items')[0]->getQuote();
+        $addedItems = $observer->getData('items');
+
+        foreach ($addedItems as $item){
+            $this->klAddedToCartItemData($quote, $item);
+        }
+    }
+
+    /**
+     * Build Added to Cart payload for current Added item
+     * @param $quote
+     * @param $addedItem
+     */
+    public function klAddedToCartItemData($quote, $addedItem)
+    {
+        $addedProduct = $addedItem->getProduct();
+        $addedItemData = array(
+            'AddedItemCategories' => (array) $this->getCategoryName($addedProduct->getCategoryIds()),
+            'AddedItemDescription' => (string) strip_tags($addedProduct->getDescription()),
+            'AddedItemImageUrlKey' => (string) $addedProduct->getData('small_image'),
+            'AddedItemPrice' => (float) $addedProduct->get_price(),
+            'AddedItemQuantity' => (int) $addedItem->getQty(),
+            'AddedItemProductID' => (int) $addedProduct->getId(),
+            'AddedItemProductName' => (string) $addedItem->getName(),
+            'AddedItemSku' => (string) $addedProduct->getSku(),
+            'AddedItemUrl' => (string) $addedProduct->getProductUrl()
+        );
+
+        $klAddedToCartPayload = array_merge(
+            $this->klBuildCartData( $quote, $addedItem ),
+            $addedItemData
+        );
+
+        // Creating a custom session variable in the checkout session since quoteId is not set at this point for guest checkouts
+        $this->getCheckoutSession()->setKlAddedToCartKey( $klAddedToCartPayload );
+    }
+
+    /**
+     * Helper function to add items from cart to Added to Cart payload
+     * @param $quote
+     * @param $addedItem
+     * @return array
+     */
+    public function klBuildCartData($quote, $addedItem)
+    {
+        $cartItems = $quote->getItems() ?? array();
```

Comment:
Note that configurable products for example are having 2 quote items, one of it is "simple" product that you selected and one of it "parent". I guess you should select the visible items only and (maybe) get some data from the parent item